src/java/org/apache/cassandra/service/AmazonCorrettoCryptoProviderImpl.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.service;
+
+import javax.crypto.Cipher;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.cassandra.exceptions.StartupException;
+import com.amazon.corretto.crypto.provider.AmazonCorrettoCryptoProvider;
+
+public class AmazonCorrettoCryptoProviderImpl implements ICryptoProvider
+{
+    private static final Logger logger = LoggerFactory.getLogger(AmazonCorrettoCryptoProviderImpl.class);
+    @Override
+    public void installProvider() throws StartupException
+    {
+        try
+        {
+            AmazonCorrettoCryptoProvider.install();
```

Comment:
Why not to install the provider in `DatabaseDescriptor.applyAll` ? E.g. before `applySeedProvider`? After it is installed, there are startup checks done. So the startup check will be done on installed provider.